tests/Unit/Pool/accelerator_get_is_peer_api.cpp

```diff
@@ -0,0 +1,34 @@
+// XFAIL: Linux
+// RUN: %hc %s -o %t.out && %t.out
+
+#include <hc.hpp>
+
+/**
+ * Test if hc::accelerator::get_is_peer() works fine.
+ * Create the default accelerator and check if others 
+ * is peer of it.
+ * accelerator is peer of itself.
+ * FIXME: on current system, dGPU is peer of any each  
+ * other, we should expect is_get_peer() return true.
+ */
```

Comment:
@facao  the unit test assume all dGPUs are peers to each other, which may be true for supported configuration. but for systems without P2P support peers.size() would be zero and this test would fail.

as the result of the test depends on the hardware configuration, and in many cases won't simply work, so I'd propose ways to improve this test case. And I'm more inclined to ask you improve the test as the 2nd method.

1. relax this test so as long as long as acc.get_is_peer() could return a valid value, then call the test pass. this would reduce the test case to merely check whether the API is available in the header file.
1. change the test case, and use HSA runtime API to query if the hardware do have HSA agents who are peers, and then check if the result of acc.get_is_peer() agree with it. You can refer to the following unit test as an example:

https://github.com/RadeonOpenCompute/hcc/blob/master/tests/Unit/AsyncPFE/accelerator_view_wait.cpp